core/src/dird/ua_restore.cc

```diff
@@ -1112,52 +1142,43 @@ static bool AskForFileregex(UaContext* ua, RestoreContext* rx)
  * should insert as
  * 0, 1, 2, 3, 4, 5, 6
  */
-static void AddDeltaListFindex(RestoreContext* rx, struct delta_list* lst)
+void AddDeltaListFindex(RestoreContext* rx, delta_list* lst)
 {
   if (lst == NULL) { return; }
   if (lst->next) { AddDeltaListFindex(rx, lst->next); }
   AddFindex(rx->bsr.get(), lst->JobId, lst->FileIndex);
 }
 
-static bool BuildDirectoryTree(UaContext* ua, RestoreContext* rx)
+void BuildDirectoryTree(UaContext* ua, RestoreContext* rx, TreeContext& tree)
```

Comment:
What happens if I call this function twice with the same tree ? Probably a memory leak.  Since we expect tree to be totally uninitialised here its probably better to create a new TreeContext in this function and then return it at the end of it.  For this you probably need to implement a move constructor for TreeContext.